mlflow/utils/requirements_utils.py

```diff
@@ -533,6 +551,14 @@ def _check_requirement_satisfied(requirement_str):
             requirement=requirement_str,
         )
 
+    if pkg_name == "mlflow" and "gateway" in req.extras:
+        try:
+            import mlflow.gateway
+        except ModuleNotFoundError:
+            return _MismatchedPackageInfo(
+                package_name=pkg_name, installed_version=None, requirement=requirement_str
+            )
```

Comment:
reusing the gateway detection logic from [here](https://github.com/mlflow/mlflow/blob/master/mlflow/__init__.py#L213-L216)

as mentioned in the PR description, i could maybe change `installed_version` here to be something like `mlflow=={version}` rather than `None`, would that be more helpful?

Reply:
The error message would look like this?

```
mlflow (current: uninstalled, requirement: mlflow[gateway])
```

Reply:
yup exactly, i'm wondering if people will get confused and think that mlflow is not installed at all

Reply:
Maybe update package_name to be mlflow[gateway]?

Reply:
> Maybe update package_name to be mlflow[gateway]?

+1